orttraining/orttraining/training_ops/cpu/tensor/gather_nd_grad.cc

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "orttraining/training_ops/cpu/tensor/gather_nd_grad.h"
+#include "core/providers/cpu/tensor/gather_nd.h"
+#include "core/common/common.h"
+namespace onnxruntime {
+
+#ifndef DISABLE_CONTRIB_OPS
+
+namespace contrib {
+ONNX_OPERATOR_KERNEL_EX(GatherNDGrad, kMSDomain, 1, kCpuExecutionProvider,
+                        KernelDefBuilder()
+                            .TypeConstraint("T", {DataTypeImpl::GetTensorType<float>(),
+                                                  DataTypeImpl::GetTensorType<double>()})
+                            .TypeConstraint("Tind", {DataTypeImpl::GetTensorType<int64_t>(),
+                                                     DataTypeImpl::GetTensorType<int32_t>()}),
+                        GatherNDGrad);
+
+}  // namespace contrib
+
+#endif
+
+template <typename InputT>
+struct GatherNDGradComputeImpl {
+  void operator()(GatherNDBase::Prepare& p, const Tensor* update_tensor) const {
+    const int64_t grad_size = update_tensor->Shape().Size();
+    const int64_t slice_size = p.element_count_per_slice;
+    const InputT* input_base_casted = reinterpret_cast<const InputT*>(p.input_base);
+    InputT* output_base_casted = reinterpret_cast<InputT*>(p.output_base);
+#ifdef _OPENMP
+#pragma omp parallel for
```

Comment:
i think we should not use #pragma omp directly, see this doc:
https://github.com/microsoft/onnxruntime/blob/master/docs/NotesOnThreading.md

Reply:
oh yeah, good point.